needy/dev.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import print_function
+
+import argparse
+import logging
+import os
+import sys
+import textwrap
+
+from .needy import ConfiguredNeedy
+
+
+def enable(args=[]):
+    parser = argparse.ArgumentParser(
+        prog='%s dev enable' % os.path.basename(sys.argv[0]),
+        description='Enables dev mode for a library. Once enabled, dev mode causes a library to always be considered out-of-date, and skips the source cleaning normally done during builds.'
+    )
+    parser.add_argument('library', help='the library to enable dev mode for')
+    parameters = parser.parse_args(args)
+
+    with ConfiguredNeedy('.', parameters) as needy:
+        needy.set_development_mode(parameters.library, True)
+
+    return 0
+
+
+def disable(args=[]):
+    parser = argparse.ArgumentParser(
+        prog='%s dev disable' % os.path.basename(sys.argv[0]),
+        description='Disables dev mode for a library.'
+    )
+    parser.add_argument('library', help='the library to disable dev mode for')
+    parameters = parser.parse_args(args)
+
+    with ConfiguredNeedy('.', parameters) as needy:
+        needy.set_development_mode(parameters.library, False)
+
+    return 0
+
+
+def status(args=[]):
+    parser = argparse.ArgumentParser(
+        prog='%s dev status' % os.path.basename(sys.argv[0]),
+        description='Shows the status of libraries that are currently in dev mode.'
+    )
+    parser.add_argument('library', nargs='?', help='a library to check the status of')
+    parameters = parser.parse_args(args)
+
+    with ConfiguredNeedy('.', parameters) as needy:
+        library_names = needy.development_mode_libraries()
+        if parameters.library:
+            if parameters.library in library_names:
+                logging.info('{} is in dev mode: {}'.format(parameters.library, needy.need_directory(parameters.library)))
+                return 0
+            else:
+                logging.info('{} is not in dev mode.'.format(parameters.library))
+                return 1
+
+        if library_names:
+            print('There {} {} librar{} in dev mode:'.format(
+                'are' if len(library_names) > 1 else 'is',
+                len(library_names),
+                'ies' if len(library_names) > 1 else 'y')
+            )
+
+            print('')
+            max_name_length = max([len(name) for name in library_names])
+            for name in library_names:
+                print('  {}{:{}} {}'.format(name, '', max_name_length - len(name) + 1, needy.need_directory(name)))
+            print('')
+
+            return 0
+
+    print('There are no libraries in dev mode.')
+    return 1
```

Comment:
I feel like the control flow here is a little weird with the "no libraries" print going outside of the with block. I would suggest this:

```python
with ConfiguredNeedy('.', parameters) as needy:
    library_names = needy.development_mode_libraries()
    if parameters.library:
        if parameters.library in library_names:
            logging.info('{} is in dev mode: {}'.format(parameters.library, needy.need_directory(parameters.library)))
            return 0
        else:
            logging.info('{} is not in dev mode.'.format(parameters.library))
            return 1

    if not library_names:
        print('There are no libraries in dev mode.')
        return 1

    print('There {} {} librar{} in dev mode:'.format(
        'are' if len(library_names) > 1 else 'is',
        len(library_names),
        'ies' if len(library_names) > 1 else 'y')
    )

    print('')
    max_name_length = max([len(name) for name in library_names])
    for name in library_names:
        print('  {}{:{}} {}'.format(name, '', max_name_length - len(name)  1, needy.need_directory(name)))
    print('')

return 0
```